src/shared/utils/utils.tsx

```diff
@@ -108,3 +111,28 @@ export function* deepFindAllByKey<T = string>(
     }
   }
 }
+
+export const addBlastLinksToFreeText = (
+  texts: string[],
+  primaryAccession: string
+) =>
+  texts.map((text) => {
+    const splitText = text.split(/(\d+-\d+)/);
+    return splitText.map((splitItem, i) => {
+      if (i % 2 === 1) {
+        const range = splitItem.split('-');
+        return (
+          <Link
+            to={getURLToJobWithData(JobTypes.BLAST, primaryAccession, {
+              start: +range[0],
+              end: +range[1],
+            })}
+            key={`${splitItem}`}
```

Comment:
If you have decided to go for the content instead of the index, then maybe no need to stringify it?